### DISCUSSION TITLE
Hot reload of pagesize property

### DISCUSSION THREAD

**Discussion by subucitrix** [OData/AspNetCoreOData#1095] [Ideas]
We are using AspNet.OData v7.7 for one of our production services which handles sizeable volume of traffic. 

We have enabled server side paging using the **EnableQueryAttribute.PageSize** with a default value of 100 and it works as expected. We are also able to change the default value by changing a config file and bouncing the server.

What we want to understand is will it be possible to **hot reload the config** value without a server restart? The reason behind this is that we have production traffic on the server at any time and don’t want to cause disruption to it with restarts.  

Request your kind guidance. If it’s doable can you please share some snippet or example that we can refer?

**Comment by xuzhg** [comment]
Might try to derive from 'EnableQueryAttribute' and override `OnActionExcuting` virtual method to retrieve the pagesize from configuration?

**Reply by subucitrix** [reply]
Tried the following:

public class MyEnableQuery : EnableQueryAttribute
{
    public override void OnActionExecuting(HttpActionContext actionContext)
    {
        var querySettings = GetQuerySettings(actionContext);

        querySettings.PageSize = 1000;

        InfrastructureLog.TraceInfo("Updating page size to {0}", querySettings.PageSize);

        InfrastructureLog.TraceInfo("fetching page size from actioncontext {0}", GetQuerySettings(actionContext).PageSize);

        base.OnActionExecuting(actionContext);
    }

    private static ODataQuerySettings GetQuerySettings(HttpActionContext actionContext)
    {
        var context = actionContext.Request.GetRequestContainer();
        return context.GetRequiredService<ODataQuerySettings>();
    }

}

The logs show a value of 1000 but in the OData API response we still see the default pagesize of 100.

Any ideas why the override is not working and how to fix it?
